### PR TITLE
Disable MSAA on feature level 10_0

### PR DESCRIPTION
**Pull request by Spodi**
Using MSAA in this constellation would otherwise crash on 10_0 hardware.

Would probably need a corresponding change in the GUI, so the user is informed about that.

Not really useful, as long the app doesn't even really work on 10_1 hardware (I can't find the reason why)

### CODE REVIEW

**Comment by Kenix3** [comment]
Is this PR still meant to be in draft?

**Comment by Spodi** [comment]
> Is this PR still meant to be in draft?

Yesnt. I dunno how useful this really is. 
This changes makes at least DirectX' debug layer shut up, when testing against feature level 10_0.
However, according to @malkierian 's testings with old Hardware, it seems we also need Compute Shader / Direct Compute support on the hardware we are running at or else SoH is just crashing on start (even without MSAA). Compute shaders / Direct compute is optional on feature level 10_0 and 10_1. (There is an example how to test for that in here somewhere: https://learn.microsoft.com/en-us/windows/win32/direct3d11/direct3d-11-advanced-stages-compute-create). I think Malkierian wanted to do *some* check against Direct Compute support and automatically switch over to OpenGL.

I don't have any old hardware to test and poke around at stuff. 

Tldr:
* SoH still crashes at start if there is no Direct Compute support on your Hardware. 
* This PR probably prevents MSAA crashes on hardware with only feature level 10_0 support, by disabling MSAA - if your hardware supports direct compute in the first place.
* there is no change in the UI telling you the MSAA slider does nothing

**Comment by Spodi** [comment]
Welp, I'm marking the Pr as ready. You decide what to do with it. Maybe someone else is patient enough to find if the port is even getting max MSAA from LUS for the MSAA slider or if this need some additional cvar or function. I'm open to suggestions.

**Comment by Kenix3** [comment]
> Welp, I'm marking the Pr as ready. You decide what to do with it. Maybe someone else is patient enough to find if the port is even getting max MSAA from LUS for the MSAA slider or if this need some additional cvar or function. I'm open to suggestions.

It's hardcoded in Ship. 
https://github.com/HarbourMasters/Shipwright/blob/develop/soh/soh/SohMenuBar.cpp#L316
I don't think LUS exposes this, but the slider is always 1-8.

1 is in the slider, and it's always a valid value. This basically just means on older hardware the slider will be ignored. Maybe make a PR to SoH with a tooltip warning.
